controller/src/main/java/io/pravega/controller/store/client/StoreClientFactory.java

```diff
@@ -49,6 +51,12 @@ public static StoreClient createZKStoreClient(CuratorFramework client) {
     }
 
     private static CuratorFramework createZKClient(ZKClientConfig zkClientConfig) {
+        if (zkClientConfig.isConnectionToZooKeeperSecure()) {
+            System.setProperty("zookeeper.client.secure", "true");
```

Comment:
Is this the only way to setup ZK SSL? I generally disagree with setting properties like this - it overrides anything that may already be there.

Reply:
It can be done via the `ZKClientConfig` rather than setting a system property:

https://github.com/apache/zookeeper/blob/faa7cec71fddfb959a7d67923acffdb67d93c953/zookeeper-client/zookeeper-client-java/src/main/java/org/apache/zookeeper/client/ZKClientConfig.java#L33

but @arvindkandhare says that it can't be done with curator (not exposed), but that I don't know.